src/providers/FileSystemProvider/FileSystemProvider.ts

```diff
@@ -259,9 +259,10 @@ export class FileSystemProvider implements vscode.FileSystemProvider {
         if (error) {
           console.log(error);
           if (error.errorText.includes(" #5540:")) {
+            const nsUpper = api.config.ns.toUpperCase();
             const message = `User '${api.config.username}' cannot list ${
-              csp ? "web application" : "namespace"
-            } contents. To resolve this, execute the following SQL in the ${api.config.ns.toUpperCase()} namespace:\n\t GRANT EXECUTE ON %Library.RoutineMgr_StudioOpenDialog TO ${
+              csp ? `web application '${uri.path}'` : "namespace"
```

Comment:
This could look odd if uri.path is "" (the root).

Reply:
I think for the root, uri.path="/"

In any case, I think this situation is very unlikely to occur. ISFS workspaces are typically created from the Server Manager tree, and a user without read permission on a namespace's code database wouldn't get to see the namespace in that tree, nor therefore the web apps (possibly "/") associated with it. The "/_vscode/XXX" path that's used to seek the .vscode/settings.json file for namespace XXX is atypical because the user doesn't browse the SM tree for it, and it comes from the "/_vscode" web app which, if defined, belongs to the %SYS namespace.